packages/contract/src/util.ts

```diff
@@ -15,25 +15,19 @@ export const getContractRoot = (bytecode: BytesLike): string => {
     chunks.push(chunk);
   }
 
-  const totalBytes = chunks.reduce((sum, chunk) => chunk.byteLength + sum, 0);
   const lastChunk = chunks[chunks.length - 1];
-  const isDivisibleBy16 = totalBytes % 16 === 0;
-  const remainingBytes = chunkSize - lastChunk.length;
-  if (!isDivisibleBy16) {
-    const nearestMultiple = Math.ceil(remainingBytes / 8) * 8;
-    const paddedChunkLength = lastChunk.length + nearestMultiple;
-    const paddedChunk = new Uint8Array(paddedChunkLength).fill(0);
-    paddedChunk.set(lastChunk, 0);
-    chunks[chunks.length - 1] = paddedChunk;
-  }
+  const remainingBytes = bytes.length % chunkSize;
+  const paddedChunkLength = remainingBytes + ((8 - (remainingBytes % 8)) % 8);
+  const newChunk = lastChunk.slice(0, paddedChunkLength);
+  chunks[chunks.length - 1] = newChunk;
 
   return calcRoot(chunks.map((c) => hexlify(c)));
 };
 
 export const getContractStorageRoot = (storageSlots: StorageSlot[]): string => {
   const tree = new SparseMerkleTree();
 
-  storageSlots.forEach(({ key, value }) => tree.update(key, value));
+  storageSlots.forEach(({ key, value }) => tree.update(sha256(key), value));
```

Comment:
@danielbate Should we add a `TODO` and a follow-up issue for doing the key-hashing from inside the Merkle classes? In the future, if we start reusing `fuel-merkle` via WASM, this would be automatically solved.

Reply:
Never mind, I've just read the "Follow ups" section from the PR description. 👍